src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -100,20 +101,43 @@ namespace {
 NetworkInfo sNetworks[kMaxNetworks];
 } // namespace
 
-// TODO: We cannot get length of operational dataset here, should be fixed after we have byte string type.
-// Note: The codegen for OCTET_STRING will be implemented later since no other clusters are using OCTET_STRING.
 EmberAfNetworkCommissioningError OnAddThreadNetworkCommandCallbackInternal(app::Command *, EndpointId,
-                                                                           const uint8_t * operationalDataset, uint64_t breadcrumb,
+                                                                           const uint8_t * operationalDataset,
+                                                                           size_t operationalDatasetLen, uint64_t breadcrumb,
                                                                            uint32_t timeoutMs)
 {
-    // TODO: Implement AddThreadNetworkCommand after OCTET_STRING is supported.
-    return EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_NETWORK_NOT_FOUND;
+    EmberAfNetworkCommissioningError err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED;
+
+    for (size_t i = 0; i < kMaxNetworks; i++)
+    {
+        if (sNetworks[i].mNetworkType == NetworkType::kUndefined)
+        {
+            VerifyOrExit(operationalDatasetLen <= kMaxThreadDatasetLen, err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
+
+            memcpy(sNetworks[i].mData.mThread.mDataset, operationalDataset, operationalDatasetLen);
+            sNetworks[i].mData.mThread.mDatasetLen = static_cast<uint8_t>(operationalDatasetLen);
```

Comment:
Why is this cast safe?

Reply:
The cast is safe since we have cheked the value before, added a few comments on this.
Also, can we make CanCastTo available to static_assert?

Reply:
CanCastTo is fundamentally a dynamic check (checking whether the given value is in some range, where the range depends on the target type), so it's hard to make it static_assertable.

We might be able to do something with multiple overloads of CanCastTo based on some template stuff, with some of them constexpr (when the source type is no larger than the target type) and some not... but in those cases when the source type is no larger we don't need `CanCastTo` to start with, because we can assign directly and the compiler won't complain.  So I'm not sure there's a point.